src/quicr_client_raw_session.cpp

```diff
@@ -687,6 +685,7 @@ QuicRClientRawSession::handle(messages::MessageBuffer&& msg)
       messages::SubscribeEnd subEnd;
       msg >> subEnd;
 
+      std::lock_guard<std::mutex> _(session_mutex);
```

Comment:
It needs to be at the subscribe/unsubscribe (api) method because insert and erase both effect the iterators.  `sub_delegates` also needs to be protected for both insert and erase.

Reply:
Actually, after looking at this again, the usages of this don't actually do anything else, so perhaps @paulej is correct, the lock should be in the `removeSubscriptionState` instead.... though it should be known that that's where the locking is happening to avoid this deadlock happening again if we ever use this method anywhere else.

Reply:
I have decided to keep the locks as Tim suggests, and leave them in the higher calling method (unsub, subEnd), since the remove method is a helper and technically has no ties to the locking mechanics (bit of a stretch to say that, but I'm sticking with it).